### PR TITLE
Multiple key=value pairs to variable create and update

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

Allow users to pass multiple key=value pairs to variable create and update

## 🎟 Issue(s)

- https://github.com/astronomer/astro-cli/issues/663

## 🧪 Functional Testing

- added unit tests
- Tested manually

## 📸 Screenshots

Screen Shot 2022-07-26 at 2 31 40 PM

## 📋 Checklist

- [X] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [X] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [X] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [X] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/deployment/deployment_variable.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func VariableList(deploymentID, variableKey, ws, envFile string, useEnvFile bool
 
 // this function modifies a deployment's environment variable object
 // it is used to create and update deployment's environment variables
-func VariableModify(deploymentID, variableKey, variableValue, ws, envFile string, useEnvFile, makeSecret, updateVars bool, client astro.Client, out io.Writer) error {
+func VariableModify(deploymentID, variableKey, variableValue, ws, envFile string, variableList []string, useEnvFile, makeSecret, updateVars bool, client astro.Client, out io.Writer) error {
```
Comment (andriisoldatenko) [inline cloud/deployment/deployment_variable.go:62]: i know it's probobly unrelated to your change ... but we need reduce amount of arguments and refactor similar places via better abstraction and structs.
It's not a blocker to merge your PR, just general idea.

cc @neel-astro

File: cloud/deployment/deployment_variable.go [hunk 2]
```diff
@@ -218,6 +221,54 @@ func addVariableFromFlag(oldKeyList []string, oldEnvironmentVariables []astro.En
...
+		pair := strings.Split(variableList[i], "=")
+		if len(pair) > 1 {
+			key = pair[0]
+			val = pair[1]
+			if key == "" || val == "" {
+				fmt.Printf("Input %s has blank key or value\n", variableList[i])
+				continue
+			}
+		} else {
+			fmt.Printf("Input %s is not a valid key value pair\n", variableList[i])
```
Comment (neel-astro) [inline cloud/deployment/deployment_variable.go:239]: might want to mention the correct format in the error response
```suggestion
			fmt.Printf("Input %s is not a valid key value pair, should be of the form key=value\n", variableList[i])
```

File: cloud/deployment/deployment_variable.go [hunk 3]
```diff
@@ -218,6 +221,54 @@ func addVariableFromFlag(oldKeyList []string, oldEnvironmentVariables []astro.En
...
+			newEnvironmentVariables[num] = newEnvironmentVariable
+		default:
+			newFileEnvironmentVariable := astro.EnvironmentVariable{
+				IsSecret: makeSecret,
+				Key:      key,
+				Value:    val,
+			}
+			newEnvironmentVariables = append(newEnvironmentVariables, newFileEnvironmentVariable)
+			fmt.Printf("adding variable %s\n", key)
+		}
```
Comment (neel-astro) [inline cloud/deployment/deployment_variable.go:267]: should this logic be better in a separate function, since the same logic is being used in multiple places now?
Your call, you can do it in a separate PR as well.

**Review by neel-astro — APPROVED** [review]
looks fine to me, have left few minor comments if you want to check it out

**Review by neel-astro — APPROVED** [review]
